### ISSUE TITLE
There is no class registered to handle Instagram requests. - Swift

### ISSUE THREAD

**Issue by chahooo** [calebd/SimpleAuth#96]
Hey Caleb,

I am getting the following error when launching in Swift.
```
Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'There is no class registered to handle Instagram requests."
```
I understand (from reading one of your answers on teamtreehouse) that SimpleAuth cannot find the files associated with handling IG authentication. I have the authorisation command in my initial view controller class but  I suspect that this has something to do with the fact that I may not have called the following right in my AppDelegate.swift
```
SimpleAuth.configuration[@"instagram"] = @{
    @"client_id" : @"CLIENT_ID",
    SimpleAuthRedirectURIKey : @"https://mysite.com/auth/instagram/callback"
};
```
I couldn't figure out a way to write that in swift so I attempted to call it in a .m and .h files and bridge those into my project like so. Forgive me for the horrible coding I've never done Obj-C in my life as I am new to coding and attempted to copy off a tutorial.
```
PhotoBombersHeader.h

#ifndef Photo_Bombers_PhotoBombersHeader_h
#define Photo_Bombers_PhotoBombersHeader_h

#import <Foundation/Foundation.h>

@interface  simpAuthConfiguration : NSObject

@property (strong, nonatomic) id someProperty;

- (void) someMethod;

@end
#endif
```
```
PhotoBombersHeaderM.m

#import <Foundation/Foundation.h>
#import <SimpleAuth/SimpleAuth.h>


@implementation simpAuthConfiguration : NSObject

// This method is the one we will use to
- (void) someMethod {
    SimpleAuth.configuration[@"instagram"] = @{
        @"client_id" : @"*******",
        SimpleAuthRedirectURIKey : @"https://www.google.com"
    };
}

@end
```

And i would then call `simpAuthConfiguration` in my AppDelegate.swift after bridging PhotoBombersHeader.h so the methods could be used in my Swift project. 

TL;DR - Basically couldn't figure out a way to translate language and tried to do it a different way and it isn't really working, any chance you could help me out? 

Cheers,
Jack

**Comment by calebd** [comment]
A missing class means you don't have the provider installed properly. You will get different errors if you haven't configured that provider. How are you installing it?

And configuring it from Swift is easy:

```swift
SimpleAuth.configuration()["instagram"] = [:]
```

**Comment by chahooo** [comment]
I'm installing it through CocoaPods

**Comment by calebd** [comment]
Are you using frameworks? Can you verify that `SimpleAuthInstagramProvider.m` is in the compile sources phase of the SimpleAuth target in your project?

**Comment by chahooo** [comment]
I can confirm that it is in the SimpleAuth pods target. I think that i just need to do
```
SimpleAuth.configuration[@"instagram"] = @{
    @"client_id" : @"CLIENT_ID",
    SimpleAuthRedirectURIKey : @"https://mysite.com/auth/instagram/callback"
};
```
correct in swift code

**Comment by calebd** [comment]
Ok. But I'm telling you, that error _is not_ thrown because the provider is not configured. It is thrown if the provider class is literally not present in your binary (https://github.com/calebd/SimpleAuth/blob/master/Pod/Core/SimpleAuth.m#L46).

Can you past the SimpleAuth line from your `Podfile`?

**Comment by chahooo** [comment]
```
platform :ios, '8.0'

pod 'SimpleAuth/Instagram'
```

**Comment by calebd** [comment]
Can you confirm that this screen looks exactly like this: 
![screen shot 2015-04-28 at 10 44 14 pm](https://cloud.githubusercontent.com/assets/1761121/7385323/2ce0191c-edf8-11e4-9c84-51c20aa22723.png)

I just did this in a test project and had no issue.

**Comment by chahooo** [comment]
I can confirm that it does, Am i meant to bridge more than just SimpleAuth.h?

**Comment by calebd** [comment]
Nope.

```swift
SimpleAuth.authorize("Instagram", completion: { responseObject, error in })
```

That `Instagram` should be lowercase.

**Comment by chahooo** [comment]
Ok thanks! that works now but it's saying my client_id and redirect_uri are nil in this line
`SimpleAuth.configuration()["instagram"] = ["72262818randomclientid8222":"https://www.example.com"]`

**Comment by calebd** [comment]
That's not how dictionaries work.

``` swift
SimpleAuth.configuration()["instagram"] = [
    "client_id": "72262818randomclientid8222",
    "redirect_uri": "https://www.example.com"
]
```

Just follow the readme.

**Comment by chahooo** [comment]
My bad, thanks for the help!
